dev/tests/integration/testsuite/Magento/Bundle/_files/product_with_multiple_options_product_with_custom_options.php

```diff
@@ -0,0 +1,119 @@
+    ->setStockData(['use_config_manage_stock' => 1, 'qty' => 100, 'is_qty_decimal' => 0, 'is_in_stock' => 1])
+    ->setPriceView(1)
+    ->setPriceType(1)
+    ->setPrice(10.0)
+    ->setShipmentType(0)
+    ->setBundleOptionsData(
+        [
+            [
+                'title' => 'Option 1',
+                'default_title' => 'Option 1',
+                'type' => 'select',
+                'required' => 1,
+                'position' => 1,
+                'delete' => '',
+            ]
+        ]
+    )->setBundleSelectionsData(
+        [
+            [
+                [
+                    'product_id' => 1,
+                    'selection_qty' => 1,
+                    'selection_can_change_qty' => 1,
+                    'delete' => '',
+                    'option_id' => 1
+                ],
+                [
+                    'product_id' => 10,
+                    'selection_qty' => 1,
+                    'selection_can_change_qty' => 1,
+                    'delete' => '',
+                    'option_id' => 1
+                ],
+                [
+                    'product_id' => 11,
+                    'selection_qty' => 1,
+                    'selection_can_change_qty' => 1,
+                    'delete' => '',
+                    'option_id' => 1
+                ],
+            ]
+        ]
+    );
+$productRepository = $objectManager->create(\Magento\Catalog\Api\ProductRepositoryInterface::class);
+
+if ($product->getBundleOptionsData()) {
+    $options = [];
+    foreach ($product->getBundleOptionsData() as $key => $optionData) {
+        if (!(bool)$optionData['delete']) {
+            $option = $objectManager->create(\Magento\Bundle\Api\Data\OptionInterfaceFactory::class)
+                ->create(['data' => $optionData]);
+            $option->setSku($product->getSku());
+            $option->setOptionId(null);
+
+            $links = [];
+            $bundleLinks = $product->getBundleSelectionsData();
+            if (!empty($bundleLinks[$key])) {
+                foreach ($bundleLinks[$key] as $linkData) {
+                    if (!(bool)$linkData['delete']) {
+                        $link = $objectManager->create(\Magento\Bundle\Api\Data\LinkInterfaceFactory::class)
+                            ->create(['data' => $linkData]);
+                        $linkProduct = $productRepository->getById($linkData['product_id']);
+                        $link->setSku($linkProduct->getSku());
+                        $link->setQty($linkData['selection_qty']);
+                        $links[] = $link;
+                    }
+                }
+                $option->setProductLinks($links);
+                $options[] = $option;
+            }
+        }
+    }
+    $extension = $product->getExtensionAttributes();
+    $extension->setBundleProductOptions($options);
+    $product->setExtensionAttributes($extension);
+}
+$product->save();
```

Comment:
Please, don't use deprecated methods in your fixtures. Use the repository or the resource model for saving the product. 

Thanks

Reply:
@rogyar ,thanks for review.
This case was fixed.